### PR TITLE
feat: defer to prod

### PR DESCRIPTION
**Pull request by saravmajestic**
## Overview
Implements defer to prod feature and 2 other modules
- added 3 configs `deferToProduction, favorState, manifestPathForDeferral` for storing the defer state and applying the same while running `dbt build`
- Bigquery cost estimate module
- Project health checker
- created common react webview panel which can be extended for other panels. Details on new web view is [here](https://github.com/AltimateAI/vscode-dbt-power-user/tree/feat/defer/webview_panels)

#### Defer
- added the extra parameters (if provided by user in insights panel) to the following commands
     - build
     - run
     - test
     - testModel

## Checklist

- [x] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

### CODE REVIEW

File: src/webview_provider/insightsPanel.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+          <meta charset="UTF-8">
+          <meta name="viewport" content="width=device-width, initial-scale=1.0">
+          <title>React App</title>
+          <link rel="stylesheet" type="text/css" href="${indexCss}">
+        </head>
+    
+        <body>
+          <div id="root"></div>
+          <script>
+            window.viewPath = "/insights";
```
Comment (saravmajestic) [inline src/webview_provider/insightsPanel.ts:120]: Define the view to be rendered in react app

File: webview_panels/src/main.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+const availableRoutes = Object.keys(AvailableRoutes);
+
+/**
+ * This method selects the route to be loaded
+ * if triggered from extension, it will use the viewPath set from extension
+ * if triggered from web page, it will use the viewPath set as query param
+ */
+const getInitialIndex = (): number => {
```
Comment (saravmajestic) [inline webview_panels/src/main.tsx:21]: get the view to be loaded based on `viewPath` variable

File: .vscode/extensions.json [hunk 3]
```diff
@@ -2,6 +2,6 @@
 	// See http://go.microsoft.com/fwlink/?LinkId=827846
 	// for the documentation about the extensions.json format
 	"recommendations": [
-		"dbaeumer.vscode-eslint"
+		"dbaeumer.vscode-eslint", "esbenp.prettier-vscode"
```
Comment (saravmajestic) [inline .vscode/extensions.json:5]: Added this extension for better dx

**Review by AdiGajbhiye — APPROVED** [review]
lgtm

**Comment by saravmajestic** [comment]
Closing infavor of https://github.com/AltimateAI/vscode-dbt-power-user/pull/867

**Pull request closed without merging**